host/src/FunctionsNetHost/Grpc/IncomingGrpcMessageHandler.cs

```diff
@@ -56,15 +56,17 @@ private async Task Process(StreamingMessage msg)
 
                     if (workerConfig?.Description is null)
                     {
-                        responseMessage.FunctionEnvironmentReloadResponse = BuildFailedEnvironmentReloadResponse(new FunctionAppPayloadNotFoundException());
+                        Logger.LogTrace($"Could not find a worker config in {envReloadRequest.FunctionAppDirectory}");
+                        responseMessage.FunctionEnvironmentReloadResponse = BuildFailedEnvironmentReloadResponse();
                         break;
                     }
 
                     // function app payload which uses an older version of Microsoft.Azure.Functions.Worker package does not support specialization.
                     if (!workerConfig.Description.CanUsePlaceholder)
                     {
-                        Logger.LogTrace("App payload uses an older version of worker package which does not support specialization.");
-                        responseMessage.FunctionEnvironmentReloadResponse = BuildFailedEnvironmentReloadResponse(new EnvironmentReloadNotSupportedException());
+                        Logger.LogTrace("App payload uses an older version of Microsoft.Azure.Functions.Worker SDK which does not support placeholder.");
+                        var e = new EnvironmentReloadNotSupportedException("This app is not using the latest version of Microsoft.Azure.Functions.Worker SDK and therefore does not leverage all performance optimizations. See https://aka.ms/azure-functions/dotnet/placeholders for more information.");
```

Comment:
It may not be super-important for this scenario, but there won't be a StackTrace with this without throwing it, will there? You can throw/catch the exception to capture the StackTrace correctly.